### PR TITLE
chore(docs): organize configurations.rst

### PR DESCRIPTION
**Pull request by erikayasuda**
Organizing our configuration docs. We should eventually switch over to using Envier for all of these so they can be auto-generated and the raw .rst file is much more clean, but for now this will help with readability. No content was changed, just reshuffling of the ordering and some grouping.

See the sample generated doc [here](https://ddtrace.readthedocs.io/en/erikayasuda-organize-configs/configuration.html) compared to what is published as [latest](https://ddtrace.readthedocs.io/en/latest/configuration.html).

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by tabgok — APPROVED** [review]
The new organization is great, makes a lot more sense to me!

File: docs/configuration.rst [hunk 1]
```diff
@@ -482,184 +424,371 @@ The following environment variables for the tracer are supported:
...
-   DD_USER_MODEL_EMAIL_FIELD:
-      type: String
-      default: ""
-      description: |
-         Field to be used to read the user email when using a custom ``User`` model for the automatic login events. This field will take precedence over automatic inference.
+      
       version_added:
          v1.15.0:
 
    DD_USER_MODEL_NAME_FIELD:
```
Comment (tushar-datadog) [inline docs/configuration.rst:782]: Quick question, how have you categorized them?

Comment (erikayasuda) [reply docs/configuration.rst:782]: I generally categorized by product, which is probably still too high-level. I'm trying to get some feedback (mainly from API SDK and ASM) about possibly bucketing these at a lower level/feature level instead (I kinda did this for sampling already).

Comment (brett0000FF) [reply docs/configuration.rst:782]: @erikayasuda -- very nice! If you're looking for some lower level categories, you could see how we recently reorganized these options for all languages on docs.datadoghq.com: https://docs.datadoghq.com/tracing/trace_collection/library_config/.

These are roughly the categories we tried to use:

- Unified service tagging
- Traces
- Agent
- ASM
- Databases
- Database monitoring
- Errors
- Logs
- OpenTelemetry
- Profiling
- Trace context propagation

**Review by brett0000FF — COMMENTED** [review]
Looks good! I left a comment with some examples of another reorg we did on our public docs if you're looking for more category ideas: https://github.com/DataDog/dd-trace-py/pull/11076/files#r1826043348.

File: docs/configuration.rst [hunk 2]
```diff
@@ -28,306 +33,244 @@ The following environment variables for the tracer are supported:
...
+      default: True
+      
+      description: |
+         Enables creation of consumer span when AWS SQS and AWS Kinesis ``poll()`` operations return no records. When disabled, no consumer span is created
+         if no records are returned.
+      
+      version_added:
+         v2.6.0:
+
+   DD_BOTOCORE_PROPAGATION_ENABLED:
```
[the remark this answers is not in the document]
Comment (erikayasuda) [reply docs/configuration.rst:90]: I did a [quick check](https://github.com/DataDog/dd-trace-py/blob/bd7cc029617ba47b89fc99b8714cb27432a556ed/ddtrace/contrib/internal/botocore/patch.py#L95-L103), and looks like `DD_BOTOCORE_DISTRIBUTED_TRACING` gets set to `botocore.distributed_tracing` and used only in tests? I see `DD_BOTOCORE_PROPAGATION_ENABLED` get set to `botocore.propagation_enabled` during patching, and this gets passed into the core dispatch call for [SQS](https://github.com/DataDog/dd-trace-py/blob/bd7cc029617ba47b89fc99b8714cb27432a556ed/ddtrace/contrib/internal/botocore/services/sqs.py#L112) and [Kinesis](https://github.com/DataDog/dd-trace-py/blob/bd7cc029617ba47b89fc99b8714cb27432a556ed/ddtrace/contrib/internal/botocore/services/kinesis.py#L108). This is out of scope for this PR, but wondering if we can consolidate this into just `DD_BOTOCORE_DISTRIBUTED_TRACING`? They're both bool configs, with the only difference being that `DD_BOTOCORE_DISTRIBUTED_TRACING` is set to `True` by default and the other one is set to `False`

[the remark this answers is not in the document]
Comment (erikayasuda) [reply docs/configuration.rst:90]: For more context, `DD_BOTOCORE_PROPAGATION_ENABLED` was added 3 months ago via this PR: https://github.com/DataDog/dd-trace-py/pull/10059

File: docs/configuration.rst [hunk 3]
```diff
@@ -10,337 +10,314 @@ see specific integration documentation for more details.
...
+     
      description: |
-         Enable sending of spans to the Agent. Note that instrumentation will still be installed and spans will be
-         generated.
+         Enables distributed tracing for the specified <INTEGRATION>.
+
      version_added:
-       v0.41.0: |
-           Formerly named ``DATADOG_TRACE_ENABLED``
+       v2.7.0:
```
Comment (erikayasuda) [inline docs/configuration.rst:52]: @mabdinur The version added here is different based on the integration. I'm thinking of changing this to just:

```suggestion
       Varies per integration.
```

Or should we list out the individual integrations here? 😅

**Review by mabdinur — APPROVED** [review]

**Pull request merged**